### PR TITLE
Edit in $EDITOR

### PR DESCRIPTION
**Pull request by let-def**
This PR allows to manipulate the log in the text editor set by user ($EDITOR or whatever xdg-open or open pick).
To be merged after #9

### CODE REVIEW

**Comment by craigfe** [comment]
I really like this improvement. 

One thought: the `` `Right `Activate`` action fails to trigger if the logs have not yet been loaded in the pane; it might be nice if the command to open the editor was buffered so that the logs will be opened as soon as they are ready. While experimenting with this branch, I often run into this issue of needing to press `l` more than once to get what I want :slightly_smiling_face:

File: src/main.ml [hunk 1]
```diff
@@ -1,488 +1,323 @@
...
+          let safe_name = Filename.quote fname in
+          let candidates =
+            match Sys.getenv_opt "EDITOR" with
+            | Some x -> [ x ]
+            | None -> (
+                match Sys.getenv_opt "PAGER" with
+                | Some x -> [ x ]
+                | None -> [] )
+          in
+          let candidates = candidates @ [ "xdg-open"; "open" ] in
```
Comment (Julow) [inline src/main.ml]: `$VISUAL` should be checked before `$EDITOR`, it should be a terminal editor too.

On macos, `open` terminates immediately, sometimes before the file is loaded by the editor.

File: src/main.ml [hunk 2]
```diff
@@ -1,488 +1,323 @@
...
+          ignore
+            (List.exists
+               (fun bin ->
+                 Sys.command (Filename.quote bin ^ " " ^ safe_name) <> 127)
+               candidates)
+        with _ -> Sys.remove fname );
```
Comment (Julow) [inline src/main.ml]: Is this reachable ?

**Comment by let-def** [comment]
I adressed both of your concerns:
- `VISUAL` is checked before `EDITOR`
- if the log has not been received yet, the command is scheduled
- now a single log file is created (and rewritten as necessary) and deleted when the software is closed

**Review by craigfe — APPROVED** [review]

**Pull request merged**